src/main/kotlin/com/zepben/evolve/cim/iec61970/base/core/NameType.kt

```diff
@@ -23,24 +23,45 @@ import com.zepben.evolve.services.common.extensions.asUnmodifiable
 class NameType(val name: String) {
 
     private var namesIndex: MutableMap<String, Name> = mutableMapOf()
-    private var namesMultiIndex: MutableMap<String, MutableList<Name>> = mutableMapOf()
+    private var namesMultiIndex: MutableMap<String, MutableMap<IdentifiedObject, Name>> = mutableMapOf()
 
     var description: String = ""
 
     /**
      * All names of this type.
      */
-    val names: Sequence<Name> get() = namesMultiIndex.values.asSequence().flatten() + namesIndex.values.asSequence()
+    val names: Sequence<Name> get() = namesMultiIndex.flatMap { it.value.values }.asSequence() + namesIndex.values.asSequence()
 
+    /**
+     * Check if the [NameType] contains a [Name].
+     *
+     * @param name the name of the required [Name] you are searching
+     * @return A [Boolean] indicating if a matching [Name] can be found in the [NameType]
+     */
     fun hasName(name: String): Boolean = namesIndex.containsKey(name) || namesMultiIndex.containsKey(name)
 
     /**
-     * Get all the [Name] instances for the provided [name].
+     * Get all the [Name] instances for the provided [IdentifiedObject].
      *
+     * @param obj the [IdentifiedObject] you are finding the [Name] for
      * @return A list of [Name]
      */
-    fun getNames(name: String): List<Name> =
-        namesIndex[name]?.let { listOf(it) } ?: namesMultiIndex[name].asUnmodifiable()
+    fun getNames(obj: IdentifiedObject): List<Name> =
+        namesIndex.values
+            .toList()
```

Comment:
todo: Why do you have the `toList()` here? If this is required for some purpose can you please document what it is , otherwise remove it as the `filter` on the `values` will already produce a `List`.

Reply:
Removed, was getting concurrent error before but not anymore.